flood_adapt/config.py

```diff
@@ -1,167 +1,85 @@
-
-    Returns
-    -------
-    None
-
-    Raises
-    ------
-    ValueError
-        If DATABASE_ROOT is not set or if the provided database_name is not a valid directory in DATABASE_ROOT.
-    """
-    if database_name == get_database_name():
-        return
-    db_root = get_database_root()
-    if db_root is None:
-        raise ValueError(
-            "DATABASE_ROOT is not set, set it before setting DATABASE_NAME\n"
-        )
-
-    full_database_path = Path(db_root, database_name)
-    if not full_database_path.is_dir():
-        raise ValueError(f"{full_database_path} is not a valid directory\n")
-
-    if get_database_name() is None:
-        os.environ["DATABASE_NAME"] = str(database_name)
-        print(f"database_name set: {database_name}")
-    elif overwrite:
-        print(f"database_name overwritten: {database_name}")
-        os.environ["DATABASE_NAME"] = str(database_name)
-
-
-def get_database_root() -> Union[Path, None]:
-    """
-    Get the root directory for the database.
-
-    Returns
-    -------
-    Path or None
-        The path to the root of the database if the DATABASE_ROOT environment variable is set,
-        None otherwise.
-    """
-    if os.environ.get("DATABASE_ROOT", None):
-        return Path(os.environ["DATABASE_ROOT"])
-
-
-def get_system_folder() -> Union[Path, None]:
-    """
-    Get the system folder path.
-
-    Returns
-    -------
-    Path or None
-        The system folder path if the SYSTEM_FOLDER environment variable is set, otherwise None.
-    """
-    if os.environ.get("SYSTEM_FOLDER", None):
-        return Path(os.environ["SYSTEM_FOLDER"])
-
-
-def get_database_name() -> Union[str, None]:
-    """
-    Get the database name.
-
-    Returns
-    -------
-    str or None
-        The database name if the DATABASE_NAME environment variable is set, otherwise None.
-    """
-    return os.environ.get("DATABASE_NAME", None)
-
-
-def parse_config(config_path: Path, overwrite: bool = True) -> dict:
-    """
-    Parse the configuration file and return the parsed configuration dictionary.
+from pydantic import Field, computed_field, model_validator
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+
+class Settings(BaseSettings):
+    model_config = SettingsConfigDict(
+        env_ignore_empty=True, validate_default=True
+    )  # empty env uses default
+
+    database_root: Path = Field(default=Path(__file__).parents[2] / "Database")
+    system_folder: Path
+    database_name: str = Field(default="default")
+    sfincs_path: Path
+    fiat_path: Path
+
+    @computed_field
+    @property
+    def database_path(self) -> Path:
+        return self.database_root / self.database_name
+
+    @model_validator(mode="before")
+    def set_defaults_if_missing(cls, data: Any) -> "Settings":
+        if isinstance(data, dict):
+            cls._validate_system_folder(data)
+            cls._validate_fiat_path(data)
+            cls._validate_sfincs_path(data)
+        return data
```

Comment:
Could you add some checks after initializing the settings here?

I think it would be nice to add a separate model_validator(mode="after") that actually checks that the computed fields for fiat_path and sfincs_path are pointing to the .exe/binaries that exist and raises if they dont.